pkg/metrics/pushgateway.go

```diff
@@ -0,0 +1,88 @@
+package metrics
+
+import (
+	"os"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/push"
+	log "github.com/sirupsen/logrus"
+)
+
+var (
+	namespace     = os.Getenv("NAMESPACE")
+	podName       = os.Getenv("POD_NAME")
+	promNamespace = "vault_creds"
+
+	errorTime = prometheus.NewGauge(prometheus.GaugeOpts{
+		Namespace: promNamespace,
+		Name:      "last_renewal_error_timestamp_seconds",
+		Help:      "The timestamp of the last error during renewal of a secret",
+	})
+
+	errorCount = prometheus.NewCounter(prometheus.CounterOpts{
+		Namespace: promNamespace,
+		Name:      "error_count",
+		Help:      "Number of errors when renewing credentials",
+	})
+
+	successTime = prometheus.NewGauge(prometheus.GaugeOpts{
+		Namespace: promNamespace,
+		Name:      "last_renewal_success_timestamp_seconds",
+		Help:      "The timestamp of the last successful renewal of a secret",
+	})
+
+	leaseExpiration = prometheus.NewGauge(prometheus.GaugeOpts{
+		Namespace: promNamespace,
+		Name:      "time_until_secrets_expire",
```

Comment:
`expiry_time_seconds` or something like that would be more prometheusy 
see https://prometheus.io/docs/practices/naming/